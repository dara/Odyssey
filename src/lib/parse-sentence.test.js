import parseSentence from './parse-sentence';

describe('parseSentence', () => {
    test('should be able to parse example 1', () => {
        const example = 'Emma is at @Moma\'s homepage. After briefly scanning the page she decides that the Explore Architecture section might be the best place to start rather than search.';
        expect(parseSentence(example)).toMatchSnapshot();
    });

    test('should be able to parse example 2', () => {
        const example = 'She sees options to view ="Feature Based Content" and also an area called ="Themes with Links" to the various themes.';
        expect(parseSentence(example)).toMatchSnapshot();
    });

    test('should be able to parse example 3', () => {
        const example = 'Emma chooses to @"Browse by Style" having seen some of the Art Deco pictures in her original search.';
        expect(parseSentence(example)).toMatchSnapshot();
    });
});
