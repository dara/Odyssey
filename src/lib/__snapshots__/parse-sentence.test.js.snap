// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseSentence should be able to parse example 1 1`] = `
Object {
  "description": "Emma is at Moma's homepage. After briefly scanning the page she decides that the Explore Architecture section might be the best place to start rather than search.",
  "orientation": "@Moma's",
  "routes": null,
}
`;

exports[`parseSentence should be able to parse example 2 1`] = `
Object {
  "description": "She sees options to view Feature Based Content and also an area called Themes with Links to the various themes.",
  "orientation": null,
  "routes": Array [
    "=\\"Feature Based Content\\"",
    "=\\"Themes with Links\\"",
  ],
}
`;

exports[`parseSentence should be able to parse example 3 1`] = `
Object {
  "description": "Emma chooses to Browse by Style having seen some of the Art Deco pictures in her original search.",
  "orientation": "@\\"Browse by Style\\"",
  "routes": null,
}
`;
